### PR TITLE
Integrate test 'kvm check' of ATSec into openQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Test steps are https://confluence.suse.com/display/CC/Evaluator+Test+Plan+SLES+15+SP2#EvaluatorTestPlanSLES15SP2-3.20KVMcheck

Relate: https://progress.opensuse.org/issues/101956
Verify run:
         https://openqa.suse.de/tests/7985156#  [aarch64]
         https://openqa.suse.de/tests/7985163#  [x86_64]
         https://openqa.suse.de/tests/7985165#  [s390x]

### CODE REVIEW

**Review by jouyingbin — APPROVED** [review]
LGTM

File: tests/security/atsec/kvm_check.pm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    if (script_run('virsh net-list --all | grep default') == 0) {
+        assert_script_run('virsh net-undefine default');
+    }
+    # Define network by parsing xml file
+    assert_script_run("virsh net-define $file");
+
+    # Check if the xml file is parsed successfully
+    validate_script_output 'virsh net-list --all | grep default', qr/default\s+inactive\s+no\s+yes/;
+
+    assert_script_run("virsh net-start default");
```
Comment (lilyeyes) [inline tests/security/atsec/kvm_check.pm:39]: s/"/'/g

File: tests/security/atsec/kvm_check.pm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    select_console 'root-console';
+
+    # Install the required packages for libvirt
+    zypper_call('in libvirt');
+    zypper_call('in wget');
+
+    # Upload the network xml file
+    my $file = 'kvm_check_network.xml';
+    assert_script_run "wget --quiet " . data_url("atsec/$file");
```
Comment (lilyeyes) [inline tests/security/atsec/kvm_check.pm:27]: s/"wget --quiet"/'wget --quiet'/g

File: tests/security/atsec/kvm_check.pm (hunk 1 above)
Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm]: fixed

**Review by lilyeyes — APPROVED** [review]
lgtm

File: tests/security/atsec/kvm_check.pm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+use testapi;
+use utils;
+
+sub run {
+    my ($self) = shift;
+
+    select_console 'root-console';
+
+    # Install the required packages for libvirt
+    zypper_call('in libvirt');
```
Comment (rfan1) [inline tests/security/atsec/kvm_check.pm:22]: Suggest change: zypper_call('in libvirt wget');

File: tests/security/atsec/kvm_check.pm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    zypper_call('in wget');
+
+    # Upload the network xml file
+    my $file = 'kvm_check_network.xml';
+    assert_script_run 'wget --quiet ' . data_url("atsec/$file");
+
+    # Checked if there is a clean environment
+    if (script_run('virsh net-list --all | grep default') == 0) {
+        assert_script_run('virsh net-undefine default');
+    }
```
Comment (rfan1) [inline tests/security/atsec/kvm_check.pm:32]: @Amrysliu I don't think you need undefine default network configuration here :) it is not start by default.

Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm:32]: It's used to make sure that. If it is not start, `undefine` will not be run.

File: tests/security/atsec/kvm_check.pm (hunk 3 above)
Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm]: fixed

Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm]: fixed

File: tests/security/atsec/kvm_check.pm (hunk 4 above)
Comment (rfan1) [reply tests/security/atsec/kvm_check.pm:31]: More comments, it if is there, we need double check it is up/online, we need destroy it before undefine?
I suggest you remove these lines or add the destroy logic :)

File: tests/security/atsec/kvm_check.pm [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    # Define network by parsing xml file
+    assert_script_run("virsh net-define $file");
+
+    # Check if the xml file is parsed successfully
+    validate_script_output 'virsh net-list --all | grep default', qr/default\s+inactive\s+no\s+yes/;
+
+    assert_script_run('virsh net-start default');
+    validate_script_output 'virsh net-list --all | grep default', qr/default\s+active\s+no\s+yes/;
```
Comment (rfan1) [inline tests/security/atsec/kvm_check.pm:39]: btw, if needed, you can add 'virsh net-autostart default', but not a must one.

File: tests/security/atsec/kvm_check.pm (hunk 4 above)
Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm]: good point. then remove the check

File: tests/security/atsec/kvm_check.pm (hunk 5 above)
Comment (Amrysliu) [reply tests/security/atsec/kvm_check.pm:35]: I prefer to follow the test steps in confluence page. :)

Comment (rfan1) [reply tests/security/atsec/kvm_check.pm:35]: Sure

**Review by rfan1 — APPROVED** [review]
LGTM

**Pull request merged**